src/napari_threedee/annotators/filament_annotator.py

```diff
@@ -0,0 +1,90 @@
+from typing import Optional
+
+import napari
+from napari.layers import Image, Points
+
+from .plane_point_annotator import PlanePointAnnotator
+from ..utils.napari_utils import add_mouse_callback_safe, remove_mouse_callback_safe
+
+
+class FilamentAnnotator(PlanePointAnnotator):
+    COLOR_CYCLE = [
+        '#1f77b4',
+        '#ff7f0e',
+        '#2ca02c',
+        '#d62728',
+        '#9467bd',
+        '#8c564b',
+        '#e377c2',
+        '#7f7f7f',
+        '#bcbd22',
+        '#17becf',
+    ]
+    FILAMENT_ID_LABEL = 'filament_id'
+
+    def __init__(
+            self,
+            viewer: napari.Viewer,
+            image_layer: Optional[Image] = None,
+            enabled: bool = False
+    ):
+        self.viewer = viewer
+        self.image_layer = image_layer
+        self.points_layer = None
+        self.enabled = enabled
+
+        self.current_filament_id: int = 0
+        self.viewer.bind_key('n', self.next_filament)
+
+        if image_layer is not None:
+            self.set_layers(self.image_layer)
+
+    @property
+    def current_filament_id(self):
+        return self._current_filament_id
+
+    @current_filament_id.setter
+    def current_filament_id(self, id: int):
+        self._current_filament_id = id
+        if self.points_layer is not None:
+            self.points_layer.current_properties = {
+                self.FILAMENT_ID_LABEL: self.current_filament_id
+            }
+
+    def next_filament(self, event=None):
+        self.current_filament_id += 1
+
+    def previous_filament(self, event=None):
+        self.current_filament_id -= 1
+
+    def _mouse_callback(self, viewer, event):
+        super()._mouse_callback(viewer, event)
+
+    def _create_points_layer(self):
```

Comment:
It looks like this is somehow being called when `self.image_layer` is `None`. Do we need a guard here?

From the CI:

```python
________________ test_something_with_viewer[filament annotator] ________________
  
  widget_name = 'filament annotator'
  make_napari_viewer = <function make_napari_viewer.<locals>.actual_factory at 0x7fdd2[48](https://github.com/napari-threedee/napari-threedee/runs/8156221401?check_suite_focus=true#step:7:49)14940>
  napari_plugin_manager = <napari.plugins._plugin_manager.NapariPluginManager object at 0x7fdd2f0e8970>
  
      @pytest.mark.parametrize("widget_name", MY_WIDGET_NAMES)
      def test_something_with_viewer(widget_name, make_napari_viewer, napari_plugin_manager):
          napari_plugin_manager.register(napari_threedee, name=MY_PLUGIN_NAME)
          viewer = make_napari_viewer()
          num_dw = len(viewer.window._dock_widgets)
  >       viewer.window.add_plugin_dock_widget(
              plugin_name=MY_PLUGIN_NAME, widget_name=widget_name
          )
  
  src/napari_threedee/_tests/test_dock_widget.py:22: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .tox/py38-linux/lib/python3.8/site-packages/napari/_qt/qt_main_window.py:699: in add_plugin_dock_widget
      wdg = _instantiate_dock_widget(Widget, self._qt_viewer.viewer)
  .tox/py38-linux/lib/python3.8/site-packages/napari/_qt/qt_main_window.py:1301: in _instantiate_dock_widget
      return wdg_cls(**kwargs)
  src/napari_threedee/annotators/qt_filament_annotator.py:10: in __init__
      super().__init__(model=FilamentAnnotator, viewer=viewer)
  src/napari_threedee/base.py:65: in __init__
      self._layer_selection_widget()
  .tox/py38-linux/lib/python3.8/site-packages/magicgui/widgets/_function_gui.py:318: in __call__
      value = self._function(*bound.args, **bound.kwargs)
  src/napari_threedee/annotators/filament_annotator.py:79: in set_layers
      self.points_layer = self._create_points_layer()
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <napari_threedee.annotators.filament_annotator.FilamentAnnotator object at 0x7fdd0ca[50](https://github.com/napari-threedee/napari-threedee/runs/8156221401?check_suite_focus=true#step:7:51)b20>
  
      def _create_points_layer(self):
          layer = Points(
  >           data=[0] * self.image_layer.data.ndim,
              ndim=self.image_layer.data.ndim,
              name='filaments',
              features={self.FILAMENT_ID_LABEL: [0]},
              face_color=self.FILAMENT_ID_LABEL,
              face_color_cycle=self.COLOR_CYCLE)
  E       AttributeError: 'NoneType' object has no attribute 'data'
  
  src/napari_threedee/annotators/filament_annotator.py:[65](https://github.com/napari-threedee/napari-threedee/runs/8156221401?check_suite_focus=true#step:7:66): AttributeError

```